### ISSUE TITLE
Issue when Configuring by Cmake

### ISSUE THREAD

**Issue by giaMSU** [openscenegraph/OpenSceneGraph#732]
When I Cmake the OSG, I got this issue
_-- Configuring incomplete, errors occurred!
See also "/home/giamsu/OpenSceneGraph/CMakeFiles/CMakeOutput.log".
See also "/home/giamsu/OpenSceneGraph/CMakeFiles/CMakeError.log"._

I check the CMakeError.log, but don't get the error. Can anyone help me please?

Change Dir: /home/giamsu/OpenSceneGraph/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/make" "cmTC_ecd5e/fast"
/usr/bin/make -f CMakeFiles/cmTC_ecd5e.dir/build.make CMakeFiles/cmTC_ecd5e.dir/build
make[1]: Entering directory '/home/giamsu/OpenSceneGraph/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_ecd5e.dir/CheckSymbolExists.c.o
/usr/bin/cc    -o CMakeFiles/cmTC_ecd5e.dir/CheckSymbolExists.c.o   -c /home/giamsu/OpenSceneGraph/CMakeFiles/CMakeTmp/CheckSymbolExists.c
Linking C executable cmTC_ecd5e
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_ecd5e.dir/link.txt --verbose=1
/usr/bin/cc      -rdynamic CMakeFiles/cmTC_ecd5e.dir/CheckSymbolExists.c.o  -o cmTC_ecd5e
CMakeFiles/cmTC_ecd5e.dir/CheckSymbolExists.c.o: In function `main':
CheckSymbolExists.c:(.text+0x1b): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_ecd5e.dir/build.make:97: recipe for target 'cmTC_ecd5e' failed
make[1]: *** [cmTC_ecd5e] Error 1
make[1]: Leaving directory '/home/giamsu/OpenSceneGraph/CMakeFiles/CMakeTmp'
Makefile:126: recipe for target 'cmTC_ecd5e/fast' failed
make: *** [cmTC_ecd5e/fast] Error 2

**Comment by giaMSU** [comment]
There are the files. Please help me to fix the issue
[CMakeError.log](https://github.com/openscenegraph/OpenSceneGraph/files/2998324/CMakeError.log)
[CMakeOutput.log](https://github.com/openscenegraph/OpenSceneGraph/files/2998325/CMakeOutput.log)

**Comment by vsg-dev** [comment]
I don't recall users report an issue like this before so there must be something specific about your setup.  The information you've provide doesn't pinpoint what has gone wrong so can't provide advice at this stage, there are just info as to some cmake tests failing.   To understand where to liook next we need more info:

  What OS?  
  What OSG version?
  What CMake version?
  What version of compile?
  What was the exact command line you used when invoking cmake?
  Have you installed OpenGL and headers?
  Have you installed pthreads headers?

**Comment by openscenegraph** [comment]
@giaMSU we can't do anything to help you without clarifications requested above.  If you can't provide these then there is nothing we can do to resolve user issues and I'll close this Issue.  If I do then you'll just need to use the osg-users mailing list/forum to report your support issues as this is correct forum tor support.

**Comment by giaMSU** [comment]
Hi there,
I will clarify these information soon. 
Best

**Comment by giaMSU** [comment]
I used Ubuntu 18.04.2 LTS
CMake version 3.10.2
OSG version 3.6.3
I don't think I installed OpenGL and headers, and pthreads, so please show me how to install them. 
Thanks

**Comment by openscenegraph** [comment]
You want is to teach you how to use your OS to install standard developer tools?  Have you not yet tried to google an answer?

I would encourage you to develop a bit more initutive, to be a programmer you really need basic skills in pulling in various project dependencies, OpenGL and Pthreads are widely used well beyond the OSG so is something you should master.  If I tell you every single step you won't develop the skills to support yourself on a wide range of topics well beyond the scope of the OSG.

**Comment by giaMSU** [comment]
Thank you for your suggestion. However, it may save time for all of us,  if you just pointed out that I need to install OpenGL and Pthreads. 
Anyway, thank you so much!

**Comment by openscenegraph** [comment]
I just had a look over the OSG webpages and README and couldn't spot mention of these dependencies, so I think something must have been lost when we moved websites a while back and we haven't spotted this was missing.  The dependencies page mentions other dependencies but not this two base ones.  I'm busy with other work right now (a Vulkan based SceneGraph).  Once I've wrapped up some work there I'll look to updating the OSG docs.
